src/config.rs

```diff
@@ -61,8 +61,68 @@ struct Root {
     clouds: Clouds,
 }
 
-fn find_config() -> Option<PathBuf> {
-    let current = Path::new("./clouds.yaml");
+/// Merge two nested serde_yaml::Mapping structs
+/// The values from src are merged into dest. Values in src override values in dest.
+fn merge_mappings(src: &serde_yaml::Mapping, dest: &mut serde_yaml::Mapping) {
+    for (src_key, src_value) in src.iter() {
+        if let Some(src_mapping) = src_value.as_mapping() {
+            if let Some(dest_value) = dest.get_mut(src_key) {
+                if let Some(dest_mapping) = dest_value.as_mapping_mut() {
+                    merge_mappings(src_mapping, dest_mapping);
+                    continue;
+                }
```

Comment:
I think we could log a warning here since we're ignoring a type mismatch between the source and the destination.

Reply:
Done!